src/foam/nanos/dig/drivers/DigFormatDriver.js

```diff
@@ -270,7 +270,18 @@ foam.CLASS({
         nu = old.fclone();
         nu.copyFrom(obj);
       }
-      return dao.put(nu);
+
+      try {
+        return dao.put(nu);
+      } catch ( ValidationException ve ) {
+        throw new DAOPutException(ve.getMessage());
+      } catch ( CompoundException ce ) {
+        var clientEx = ce.getClientRethrowException();
+        if ( clientEx instanceof ValidationException ) {
+          throw new DAOPutException(clientEx.getMessage());
```

Comment:
here too?
```suggestion
          throw new DAOPutException(clientEx.getMessage(), clientEx);
```

Reply:
Fixed in https://github.com/foam-framework/foam2/pull/4772/commits/5d03767cfca7ad2e108716f8a94212e953cb0dca.